### PR TITLE
secure_uploads settings for Django

### PR DESCRIPTION
**Pull request by evgkirov**
Related issue #262 

Enabling `secure_uploads` will expose signatures to all `FileField`s, not only limited to Django admin. While this may be desired in some cases, it also poses a security risk for websites with user uploads. This has been highlighted in the documentation, but it may need to be rephrased somehow (@rsedykh?).

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -6,6 +6,16 @@ The format is based on [Keep a
 Changelog](https://keepachangelog.com/en/1.0.0/), and this project
 adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [4.1.5](https://github.com/uploadcare/pyuploadcare/compare/v4.1.4...v4.1.5) - 2023-10-30
```
Comment (rsedykh) [inline HISTORY.md:9]: Mix it into 4.2 update?

File: docs/django-widget.rst [hunk 2]
```diff
@@ -56,9 +56,21 @@ widget url:
 
     UPLOADCARE = {
         # ...
-        'upload_base_url' = 'http://path.to/your/upload/handler',
+        'upload_base_url': 'http://path.to/your/upload/handler',
     }
 
+If you have signed uploads enabled in your Uploadcare project, widget-based uploads will fail unless you enable the ``signed_uploads`` setting in your Django project.
+
+Note that in this case, the benefits of signed uploads are essentially negated for user uploads. So enabling this feature is only feasible if, for example, files are only uploaded via the admin interface.
```
Comment (rsedykh) [inline docs/django-widget.rst:64]: Maybe write this note in the changelog too?

**Review by rsedykh — CHANGES_REQUESTED** [review]
I'm not sure that I'm getting the problem, but I left my comments regarding your question.

File: HISTORY.md (hunk 1 above)
Comment (evgkirov) [reply HISTORY.md:9]: Perhaps. Depends on when and in which order these two pull requests will be merged.

File: docs/django-widget.rst (hunk 2 above)
Comment (dmitry-mukhin) [reply docs/django-widget.rst:64]: the note is confusing and I believe incorrect.
public key leak is still a thing.

**Review by dmitry-mukhin — CHANGES_REQUESTED** [review]

File: docs/django-widget.rst (hunk 2 above)
Comment (evgkirov) [reply docs/django-widget.rst:64]: Ok, need to rephrase it. At the moment I have no idea how exactly.

**Comment by dmitry-mukhin** [comment]
just get rid of it

File: HISTORY.md [hunk 3]
```diff
@@ -6,6 +6,16 @@ The format is based on [Keep a
...
+
+### Added
+
+- For `Uploadcare`:
+  - added `generate_upload_signature` method. This shortcut could be useful for signed uploads from your website's frontend where the signature needs to be passed outside of your site's Python part (e.g. for an upload widget).
```
Comment (rsedykh) [inline HISTORY.md:14]:
```suggestion
  - added `generate_upload_signature` method. This shortcut could be useful for signed uploads from your website's frontend where the signature needs to be passed outside of your website's Python part (e.g. for the uploading widget).
```
?

File: HISTORY.md [hunk 4]
```diff
@@ -6,6 +6,16 @@ The format is based on [Keep a
...
+
+- For Django integration:
+  - added `signed_uploads` setting for Django projects. When enabled, this setting exposes the generated signature to the upload widget.
```
Comment (rsedykh) [inline HISTORY.md:17]:
```suggestion
  - added `signed_uploads` setting for Django projects. When enabled, this setting exposes the generated signature to the uploading widget.
```
?

**Review by rsedykh — CHANGES_REQUESTED** [review]

**Review by rsedykh — APPROVED** [review]

**Review by dmitry-mukhin — APPROVED** [review]

**Pull request merged**